### PR TITLE
fixed image bounds error status null bug + test case

### PR DESCRIPTION
**Pull request by cato-o**
**Summarize your change.**

Added null pointer check for the error status pointer in the available image bounds function.

**Reference associated tests.**
Added two new test cases to test_clip.cpp called "test_error_ptr_null" and "test_error_ptr_null_no_media". One test has a media reference, the other test has no media reference.

### CODE REVIEW

File: tests/test_clip.cpp [hunk 1]
```diff
@@ -260,6 +260,107 @@ main(int argc, char** argv)
...
+
+        // check that if null ptr, nothing happens
+        otio::ErrorStatus* null_test = nullptr;
+
+        assertEqual(
+            mr_clip->available_image_bounds(null_test), std::optional<IMATH_NAMESPACE::Box2d>()
+        );
+
+
+
```
Comment (jminor) [inline tests/test_clip.cpp:321]: Can you remove these blank lines?

**Comment by darbyjohnston** [comment]
Hi, and thanks for the fix!

It would also be nice to fix the `available_image_bounds()` function declarations, it looks like they are missing the default `nullptr` value. We generally default the `error_status` argument to `nullptr` so there is a bit less code to write if the developer does not want to check the error status.

It looks like the declarations need to be fixed in `clip.h` and `composable.h`, just by changing:
```
std::optional<IMATH_NAMESPACE::Box2d>
    available_image_bounds(ErrorStatus* error_status) const;
```
To:
```
std::optional<IMATH_NAMESPACE::Box2d>
    available_image_bounds(ErrorStatus* error_status = nullptr) const;
```

File: tests/test_clip.cpp [hunk 2]
```diff
@@ -260,6 +260,104 @@ main(int argc, char** argv)
...
+        );
+    });
+
+    tests.add_test("test_error_ptr_null_no_media", [] {
+        using namespace otio;
+
+        // tests for no image bounds and no media reference on clip
+
+        otio::ErrorStatus              status;
+        SerializableObject::Retainer<> so =
```
Comment (darbyjohnston) [inline tests/test_clip.cpp:327]: I think it is OK to construct the `Clip` from JSON here, since it looks like it creates one with an empty media reference. I'm not sure the C++ API allows you to set an empty `media_reference`.

Out of curiosity, was there a bug caused by having a clip without a media reference?

Comment (cato-o) [reply tests/test_clip.cpp:327]: There wasn't a specific bug but I wrote this to try and get into and test the `if (!active_media)` code path. I'm not certain if it's ever possible to get there though since I think if there's an empty `media_reference`, it gets checked for and set to DEFAULT_MEDIA instead.

**Comment by cato-o** [comment]
Hi Darby! I made the changes you mentioned and it's ready for re-review! Thank you :)

**Comment by darbyjohnston** [comment]
Thanks for the changes! Just one more small change and I think it is ready; in the second test, you could use `new` to create the clip since you are setting the media references to empty:
```
SerializableObject::Retainer<Clip> clip(new Clip);
```
Or keep the JSON and remove the part about setting the media references to empty, they both do the same thing and only one is needed.

**Comment by darbyjohnston** [comment]
Looks good, thanks again!
